mod_ci/controllers.py

```diff
@@ -266,7 +269,12 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
         # Create XML file for test
         file_name = '{name}.xml'.format(name=category.name)
         single_test = etree.Element('tests')
+        check_write = False
         for regression_test in category.regression_tests:
+            if regression_test.id not in regression_ids:
+                break
```

Comment:
This seems a bit excessive to me.

What happens if you select test 2, 3 if a category holds 1, 2, 3? If my logic isn't misguiding me, it'd break out of the loop for all the remainders.

Continue seems more appropriate here.